indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -203,6 +203,8 @@ public class TaskGroup
     final String baseSequenceName;
     DateTime completionTimeout; // is set after signalTasksToFinish(); if not done by timeout, take corrective action
 
+    Boolean shutdownEarly = false; // set by SupervisorManager.stopTaskGroupEarly
```

Comment:
```suggestion
    boolean shutdownEarly = false; // set by SupervisorManager.stopTaskGroupEarly
```

^ easier if this is a primitive, then we don't need to think about handling the null state